cou/cli.py

```diff
@@ -59,11 +65,22 @@ def setup_logging(log_level: str = "INFO") -> None:
         fmt="%(asctime)s [%(levelname)s] %(message)s", datefmt="%Y-%m-%d %H:%M:%S"
     )
     root_logger = logging.getLogger()
-    root_logger.setLevel(log_level)
-    if not root_logger.hasHandlers():
-        console_handler = logging.StreamHandler()
-        console_handler.setFormatter(log_formatter)
-        root_logger.addHandler(console_handler)
+    root_logger.setLevel("DEBUG")
+
+    # handler for the log file. Log level is DEBUG
+    pathlib.Path(COU_DIR_LOG).mkdir(parents=True, exist_ok=True)
+    log_file_handler = logging.handlers.TimedRotatingFileHandler(
+        f"{COU_DIR_LOG}/cou.log", when="D", interval=1
+    )
+    log_file_handler.setFormatter(log_formatter)
```

Comment:
not-blocking: Would it make any difference to use `log_file_handler.setLevel("DEBUG")` instead of `root_logger.setLevel("DEBUG")`?

Reply:
The [logger](https://docs.python.org/3/howto/logging.html#logging-flow) acts as first filter before passing the event to the handlers. So if the logger level is at higher severity (like INFO) it will filter the event and the handler won't receive it, even if the event was matching the handler's level (DEBUG).